odl/tomo/geometry/conebeam.py

```diff
@@ -285,46 +1014,234 @@ class CircularConeFlatGeometry(HelicalConeFlatGeometry):
+        for key in ('pitch', 'pitch_offset'):
+            if key in kwargs:
+                raise TypeError('got an unexpected keyword argument {!r}'
+                                ''.format(key))
+
         super().__init__(apart, dpart, src_radius, det_radius, pitch=0,
                          axis=axis, **kwargs)
 
+    @classmethod
+    def frommatrix(cls, apart, dpart, src_radius, det_radius, init_matrix):
+        """Create an instance of `CircularConeFlatGeometry` using a matrix.
+
+        This alternative constructor uses a matrix to rotate and
+        translate the default configuration. It is most useful when
+        the transformation to be applied is already given as a matrix.
+
+        Parameters
+        ----------
+        apart : 1-dim. `RectPartition`
+            Partition of the parameter interval.
+        dpart : 2-dim. `RectPartition`
+            Partition of the detector parameter set.
+        src_radius : nonnegative float
+            Radius of the source circle.
+        det_radius : nonnegative float
+            Radius of the detector circle.
+        init_matrix : `array_like`, shape ``(3, 3)`` or ``(3, 4)``, optional
+            Transformation matrix whose left ``(3, 3)`` block is multiplied
+            with the default ``det_pos_init`` and ``det_axes_init`` to
+            determine the new vectors. If present, the fourth column acts
+            as a translation after the initial transformation.
+            The resulting ``det_axes_init`` will be normalized.
+
+        Returns
+        -------
+        geometry : `CircularConeFlatGeometry`
+            The resulting geometry.
+
+        Examples
+        --------
+        Map unit vectors ``e_y -> e_z`` and ``e_z -> -e_y``, keeping the
+        right-handedness:
+
+        >>> apart = odl.uniform_partition(0, 2 * np.pi, 10)
+        >>> dpart = odl.uniform_partition([-1, -1], [1, 1], (20, 20))
+        >>> matrix = np.array([[1, 0, 0],
+        ...                    [0, 0, -1],
+        ...                    [0, 1, 0]])
+        >>> geom = CircularConeFlatGeometry.frommatrix(
+        ...     apart, dpart, src_radius=5, det_radius=10, init_matrix=matrix)
+        >>> geom.axis
+        array([ 0., -1.,  0.])
+        >>> geom.det_refpoint(0)
+        array([  0.,   0.,  10.])
+        >>> geom.det_axes_init
+        (array([ 1.,  0.,  0.]), array([ 0., -1.,  0.]))
+
+        Adding a translation with a fourth matrix column:
+
+        >>> matrix = np.array([[0, 0, -1, 0],
+        ...                    [0, 1, 0, 1],
+        ...                    [1, 0, 0, 1]])
+        >>> geom = CircularConeFlatGeometry.frommatrix(
+        ...     apart, dpart, src_radius=5, det_radius=10, init_matrix=matrix)
+        >>> geom.translation
+        array([ 0.,  1.,  1.])
+        >>> geom.det_refpoint(0)  # (0, 10, 0) + (0, 1, 1)
+        array([  0.,  11.,   1.])
+        """
+        # Get transformation and translation parts from `init_matrix`
```

Comment:
Why does this not simply call the superclass version of this somehow? Should be able to call the superclass thing but with this class, or I may be wrong?